docs/source/install.rst

```diff
@@ -0,0 +1,294 @@
+options will include paths to system packages so PETSc can correctly find and
+link against them. If you are not then you should pass the ``--no-package-manager``
+flag to obtain a set of configure options where ``firedrake-configure``
+pessimistically assumes that no external packages are available, and hence need
+to be downloaded and compiled from source::
+
+   $ ./configure $(python3 ../firedrake-configure --no-package-manager --show-petsc-configure-options)
+
+For the ``default`` ARCH, running ``firedrake-configure`` with
+``--no-package-manager`` will produce the flags:
+
+.. literalinclude:: petsc_configure_options.txt
+   :language: text
+
+
+Customising the PETSc installation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Since ``firedrake-configure`` only outputs a string of options it is straightforward
+to make changes to the options passed to PETSc ``configure``. You can either:
+
+* Append additional options when ``configure`` is invoked::
+
+   $ ./configure $(python3 ../firedrake-configure --show-petsc-configure-options) --download-exotic-package
+
+* Write the output of ``firedrake-configure`` to a file than can be modified::
+
+   $ python3 ../firedrake-configure --show-petsc-configure-options > my_configure_options.txt
+   <edit my_configure_options.txt>
+   $ ./configure $(cat my_configure_options.txt)
+
+
+.. _install_firedrake:
+
+Installing Firedrake
+--------------------
+
+Now that the right system packages are installed and PETSc is built we can now
+install Firedrake. To do this perform the following steps:
+
+#. Set any necessary environment variables. This can be achieved using
+   ``firedrake-configure``::
+
+     $ export $(python3 firedrake-configure --show-env)
+
+   At a minimum this will set the following variables:
+
+   .. code-block:: text
+
+      PETSC_DIR=/path/to/petsc PETSC_ARCH=arch-firedrake-{default,complex} HDF5_MPI=ON
+
+#. Install Firedrake::
+
+     $ pip install --no-binary h5py git+https://github.com/firedrakeproject/firedrake.git
+
+#. Firedrake is now installed and ready for use!
+
+.. note::
+   During the installation Firedrake will compile and install petsc4py_. If
+   you have previously installed petsc4py on your computer with a different
+   PETSc then ``pip`` will erroneously reuse the existing petsc4py which is 
+   linked against the wrong library. To avoid this you need to run the
+   command::
+
+       pip cache remove petsc4py
+
+   Equivalent commands may also be necessary for mpi4py and h5py if you are
+   changing the MPI and/or HDF5 libraries in use.
+
+.. note::
+   If you are using an MPI installed into a nonstandard location it may be
+   necessary to set some additional environment variables before installation:
+
+   * ``CC`` and ``MPICC`` to the location of ``mpicc``
+   * ``CXX`` to the location of ``mpicxx``
+   * ``MPI_HOME`` to the base directory of the MPI installation (e.g. ``/usr``
+     or ``/opt/mpich``)
```

Comment:
no zenodo except for editable installs. This will be fixed in April/May for release versions of Firedrake.

Reply:
done